actionpack/lib/action_dispatch/http/cache.rb

```diff
@@ -191,7 +195,7 @@ def set_conditional_cache_control!(cache_control)
           control.merge! cache_control
 
           if control.empty?
-            self._cache_control = DEFAULT_CACHE_CONTROL
+            # Let middleware handle default behavior
```

Comment:
Given that the only time we care to set a default value here is if we already have an ETag set, I could replace this no-op with something like:

``` ruby
self._cache_control = DEFAULT_CACHE_CONTROL if etag? || last_modified?
```

I only went with the fully decoupled approach (two separate methods) since it seemed like the concerns of each were kind of different.